<?xml version="1.0" encoding="UTF-8"?>
<launch>
<arg name = "pcl_topic" default = "/zed/zed_node/point_cloud/cloud_registered"/>
    
<node pkg="tf" type="static_transform_publisher" name="yy" args="0 0 0 0 0 0 1 world map 100" />

<node pkg="px4_code" type="poseStamped2tf_node" name="pose2tf" output="screen">
    <param name="map_frame" value="/map"/>
    <param name="base_frame" value="/zed_left_camera_frame"/>
    <param name="input_pose_topic" value="/mavros/vision_pose/pose"/>
    <param name="pointcloud_topic" value="$(arg pcl_topic)"/>
</node>

	<node pkg="octomap_server" type="octomap_tracking_server_node" name="octomap_talker" output="screen" args="">
		<param name="resolution" value="0.1" />
		<param name="frame_id" type="string" value="map" />
		<param name="sensor_model/max_range" value="8.0" />
		<param name="sensor_model/miss" value="0.3" />
		<param name="save_directory" value="$(optenv OCTOMAP_SAVE_DIR ./)" />
		<param name="track_changes" value="true"/>
		<param name="listen_changes" value="false"/>
		<param name="topic_changes" value="/octomap_tracking_server/changeset" />
		<!-- added by JBS -->
		<param name="pointcloud_max_z" value="3.0" />
		<param name="pointcloud_min_z" value="-0.5" />
		<param name="color/a" value="0.5" />

		<param name="filter_ground" value="false" />
		<param name="use_sim_time" value="true" />
        <param name="min_change_pub" value="0" />
		<remap from="cloud_in" to="$(arg pcl_topic)_sync" />
	</node>

<node name="rviz" pkg="rviz" type="rviz" args="-d $(find px4_code)/rviz/logger.rviz">
</node>

</launch>
